File: 206-reverse-linked-list/reverse-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        if(!head) return head;
        ListNode* temp=head;
        ListNode* temp2=NULL;
        head=head->next;
        temp->next=temp2;
        while(head){
            temp2=head;
            head=head->next;
            temp2->next=temp;
            temp=temp2;
        }
        return temp;
    }
};

File: 206-reverse-linked-list/reverse-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        ListNode temp;
        ListNode rev = null;
        while(head != null){
            temp = head.next;
            head.next = rev;
            rev = head;
            head = temp;
        }
        return rev;
    }
}